app/containers/ReactionsList.tsx

```diff
@@ -0,0 +1,138 @@
+	userItemContainer: { marginHorizontal: 10, marginVertical: 5, flexDirection: 'row' },
+	usernameContainer: { marginHorizontal: 10, justifyContent: 'center' },
+	usernameText: { fontSize: 17, ...sharedStyles.textMedium }
+});
+const standardEmojiStyle = { fontSize: 20, color: '#fff' };
+const customEmojiStyle = { width: 25, height: 25 };
+
+interface ISharedFields {
+	baseUrl: string;
+	getCustomEmoji: TGetCustomEmoji;
+}
+
+interface IReactionsListProps extends ISharedFields {
+	message?: TMessageModel;
+}
+
+interface ITabBarItem extends ISharedFields {
+	tab: IReaction;
+	index: number;
+	activeTab?: number;
+	goToPage?: (index: number) => void;
+}
+interface IReactionsTabBar extends ISharedFields {
+	activeTab?: number;
+	tabs?: IReaction[];
+	goToPage?: (index: number) => void;
+}
+
+const TabBarItem = React.memo(({ tab, index, goToPage, activeTab, baseUrl, getCustomEmoji }: ITabBarItem) => {
+	const { colors } = useTheme();
+	return (
+		<TouchableOpacity
+			activeOpacity={0.7}
+			key={tab.emoji}
+			onPress={() => {
+				goToPage?.(index);
+			}}>
+			<View style={styles.tabBarItem}>
+				<Emoji
+					content={tab.emoji}
+					standardEmojiStyle={standardEmojiStyle}
+					customEmojiStyle={customEmojiStyle}
+					baseUrl={baseUrl}
+					getCustomEmoji={getCustomEmoji}
+				/>
+				<Text style={[styles.reactionCount, { color: colors.auxiliaryTintColor }]}>{tab.usernames.length}</Text>
+			</View>
+			<View
+				style={[
+					styles.tabLine,
+					activeTab === index
+						? { backgroundColor: colors.tintColor, height: 2 }
+						: {
+								backgroundColor: colors.separatorColor,
+								height: 1
+						  }
+				]}
+			/>
+		</TouchableOpacity>
+	);
+});
+
+const ReactionsTabBar = React.memo(({ tabs, ...props }: IReactionsTabBar) => (
+	<View>
+		<FlatList
+			data={tabs}
+			keyExtractor={item => item.emoji}
+			horizontal
+			showsHorizontalScrollIndicator={false}
+			renderItem={({ item, index }) => <TabBarItem tab={item} index={index} {...props} />}
+		/>
+	</View>
+));
+
+const UsersList = React.memo(({ tabLabel }: { tabLabel: IReaction }) => {
+	const { colors } = useTheme();
+	const { emoji, usernames } = tabLabel;
+	return (
+		<View>
+			<View style={styles.emojiName}>
+				<Text style={{ color: colors.auxiliaryTintColor }}>{emoji}</Text>
+			</View>
+			<FlatList
+				data={usernames}
+				renderItem={({ item }: { item: string }) => (
+					<View style={styles.userItemContainer}>
+						<Avatar text={item} size={35} />
+						<View style={styles.usernameContainer}>
+							<Text style={[styles.usernameText, { color: colors.titleText }]}>{item}</Text>
+						</View>
+					</View>
+				)}
+			/>
+		</View>
+	);
+});
+
+const ReactionsList = React.memo(({ message, ...props }: IReactionsListProps) => {
+	// sorting reactions in descending order on the basic of number of users reacted
+	const sortedReactions = message?.reactions?.sort(
+		(reaction1, reaction2) => reaction2.usernames.length - reaction1.usernames.length
+	);
```

Comment:
I'm not sure how you want to memoize this using usernames as `reactions` is an array in which each element is an object containing id, emoji and usernames array. It won't help until `reactions` array itself is memoized, right? Am I missing any point?

Reply:
The use memo here doesn't really have any important impact, since this is the parent component, the one that renders several times is the child, this function is only executed once when assembling the whole component.